### PR TITLE
SW-3544 Temporary monitoring plot selection

### PR DESCRIPTION
**Pull request by sgrimm**
Implement the selection logic for temporary monitoring plots, and show example
plots on the site map in the admin UI.

### CODE REVIEW

**Comment by sgrimm** [comment]
Current dependencies on/for this PR:
* main
  * **PR #1127** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1127" target="_blank">Graphite</a>  👈
    * **PR #1129** <a href="https://app.graphite.dev/github/pr/terraware/terraware-server/1129" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/terraware/terraware-server/1127?utm_source=stack-comment).

File: src/test/kotlin/com/terraformation/backend/tracking/model/PlantingZoneModelTest.kt [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+                .toSet()
+
+        val numChosenPerSubzone = availablePlotIds.map { ids -> ids.intersect(chosenIds).size }
+
+        assertEquals(listOf(1, 2, 2), numChosenPerSubzone, "Number of plots chosen in each subzone")
+      }
+    }
+
+    @Test
+    fun `throws exception if temporary plot count not set`() {
```
Comment (karthikbtf) [inline src/test/kotlin/com/terraformation/backend/tracking/model/PlantingZoneModelTest.kt:113]: I probably missed this earlier, will setting 0 as the temporary plot count throw an exception?

**Review by karthikbtf — APPROVED** [review]
Looks good to me, I can follow the logic/math here.

File: src/test/kotlin/com/terraformation/backend/tracking/model/PlantingZoneModelTest.kt (hunk 1 above)
Comment (sgrimm) [reply src/test/kotlin/com/terraformation/backend/tracking/model/PlantingZoneModelTest.kt:113]: It's valid to not have any temporary monitoring plots, just permanent ones, though admittedly this is still a topic that's being discussed on the PRD.

**Comment by sgrimm** [comment]
Updated this to take into account which subzones have had outplanting nursery withdrawals, per discussion on the PRD, to eliminate the corner case where an observation has monitoring plots but all of them are in subzones that haven't been planted yet.

File: src/main/kotlin/com/terraformation/backend/tracking/db/PlantingSiteStore.kt [hunk 2]
```diff
@@ -83,6 +84,20 @@ class PlantingSiteStore(
...
 
+  fun fetchPlantedSubzoneIds(plantingSiteId: PlantingSiteId): Set<PlantingSubzoneId> {
+    requirePermissions { readPlantingSite(plantingSiteId) }
+
+    return dslContext
+        .select(PLANTING_SUBZONES.ID)
+        .from(PLANTING_SUBZONES)
+        .where(PLANTING_SUBZONES.PLANTING_SITE_ID.eq(plantingSiteId))
+        .andExists(
+            DSL.selectOne()
```
Comment (karthikbtf) [inline src/main/kotlin/com/terraformation/backend/tracking/db/PlantingSiteStore.kt]: skip reassignments?

**Review by karthikbtf — APPROVED** [review]

File: src/main/kotlin/com/terraformation/backend/tracking/db/PlantingSiteStore.kt (hunk 2 above)
Comment (sgrimm) [reply src/main/kotlin/com/terraformation/backend/tracking/db/PlantingSiteStore.kt]: Good catch. Forgot about reassignments.

**Pull request merged**